brave/src/main/java/brave/Tracer.java

```diff
@@ -44,7 +44,35 @@
  * @see Span
  * @see Propagation
  */
-public final class Tracer {
+public final class Tracer implements Closeable {
+  // volatile for visibility on get. writes guarded by Tracer.class
+  static volatile Tracer current = null;
+
+  /**
+   * Returns the most recent tracer that hasn't been closed.
+   *
+   * <p>This object should not be cached.
+   */
+  @Nullable public static Tracer current() {
```

Comment:
Is it expected that instrumentation that uses this would just perform a delayed retry if `Tracer.current()` was null?  Its probably overkill but maybe some kind of callback approach would be better better suited as apposed to null.